syntax = "proto3";

package aegis;

service AegisGateway {
  rpc Process(AegisRequest) returns (AegisResponse);
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

message HealthCheckRequest {}
message HealthCheckResponse {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
  }
  ServingStatus status = 1;
}

message AegisRequest {
  // Зашифрованное тело запроса (содержит реальный метод, путь, заголовки и тело)
  bytes encrypted_payload = 1;

  // Метаданные (trace_id, etc), используются как "associated data".
  // НЕ ИСПОЛЬЗОВАТЬ ДЛЯ ПЕРЕДАЧИ ПУТИ/МЕТОДА (MTD violation).
  map<string, string> metadata = 2;

  // Публичный ключ клиента для идентификации сессии
  bytes public_key = 3;
}

message AegisResponse {
  // Камуфляжный HTTP-статус (для обмана внешних наблюдателей)
  int32 fake_http_status = 1;

  // Зашифрованный ответ (содержит реальный статус, заголовки и тело)
  bytes encrypted_payload = 2;

  // Метаданные ответа, используются как "associated data"
  map<string, string> metadata = 3;
}
